src/ASM/LR/ASMu3D.C

```diff
@@ -1917,9 +1917,50 @@ bool ASMu3D::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 }
 
 
+std::vector<int> ASMu3D::getEdgeNodes (int edge, int basis) const
+{
+  size_t ofs = 1;
+  for (int i = 1; i < basis; i++)
+    ofs += this->getNoNodes(i);
+
+  std::vector<LR::Basisfunction*> edgeFunctions;
+  this->getBasis(basis)->getEdgeFunctions(edgeFunctions,
+                                          static_cast<LR::parameterEdge>(edge));
+
+  std::vector<int> result(edgeFunctions.size());
+  std::transform(edgeFunctions.begin(), edgeFunctions.end(), result.begin(),
+                 [ofs](LR::Basisfunction* a) { return a->getId()+ofs; });
+
+  return result;
+}
+
+
 void ASMu3D::getBoundaryNodes (int lIndex, IntVec& nodes, int basis) const
 {
-  // TODO: Implement this before attempting FSI simulations with LR B-splines
+  if (basis == 0)
+    basis = 1;
+
+  if (!this->getBasis(basis)) return; // silently ignore empty patches
+
+  LR::parameterEdge edge;
+  switch (lIndex) {
+  case 1: edge = LR::WEST; break;
+  case 2: edge = LR::EAST; break;
+  case 3: edge = LR::SOUTH; break;
+  case 4: edge = LR::NORTH; break;
+  case 5: edge = LR::BOTTOM; break;
+  case 6: edge = LR::TOP; break;
+  default: return;
+  }
+
+  nodes = this->getEdgeNodes(edge, basis);
+
```

Comment:
Randa av et 3D element bastår av Faces og ikke Edges, så denne metoden burde vel hete getFaceNodes (og ikke getEdgeNodes) for å unngå forvirring, eller er det bare funksjonene som ligger lags en kant som returneres. Der er i såfall feil funksjonalitet.